File: frontend/src/types/watchlist-center.ts
export type WatchlistCenterSummary = {
  total_count: number;
  focus_count: number;
  normal_count: number;
  theme_resonance_count: number;
  opportunity_linked_count: number;
  active_alert_count: number;
  holding_linked_count: number;
};

export type WatchlistCenterItem = {
  ticker: string;
  display_name: string;
  watchlist_name: string;
  theme_name: string | null;
  status: string;
  reason: string;
  tradeability_state: string;
  expectation_gap_level: string;
  role_label: string;
  trend_quality: string;
  latest_price: string | null;
  change_percent: number | null;
  has_theme_resonance: boolean;
  has_opportunity_link: boolean;
  has_active_alert: boolean;
  has_holding: boolean;
  holding_action: string | null;
  linked_candidate_state: string | null;
  linked_judge_action: string | null;
  observation_priority: string;
  quick_note: string;
};

export type WatchlistCenterOverview = {
  generated_at: string;
  available: boolean;
  empty_message: string | null;
  summary: WatchlistCenterSummary;
  items: WatchlistCenterItem[];
  grouped: {
    focus_items: WatchlistCenterItem[];
    resonance_items: WatchlistCenterItem[];
    normal_items: WatchlistCenterItem[];
    holding_linked_items: WatchlistCenterItem[];
  };
};

File: python/valuecell/server/api/schemas/watchlist_center.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class WatchlistCenterSummaryData(BaseModel):
    total_count: int
    focus_count: int
    normal_count: int
    theme_resonance_count: int
    opportunity_linked_count: int
    active_alert_count: int
    holding_linked_count: int


class WatchlistCenterItemData(BaseModel):
    ticker: str
    display_name: str
    watchlist_name: str
    theme_name: str | None = None
    status: str
    reason: str
    tradeability_state: str
    expectation_gap_level: str
    role_label: str
    trend_quality: str
    latest_price: str | None = None
    change_percent: float | None = None
    has_theme_resonance: bool
    has_opportunity_link: bool
    has_active_alert: bool
    has_holding: bool
    holding_action: str | None = None
    linked_candidate_state: str | None = None
    linked_judge_action: str | None = None
    observation_priority: str
    quick_note: str


class WatchlistCenterGroupedData(BaseModel):
    focus_items: list[WatchlistCenterItemData] = Field(default_factory=list)
    resonance_items: list[WatchlistCenterItemData] = Field(default_factory=list)
    normal_items: list[WatchlistCenterItemData] = Field(default_factory=list)
    holding_linked_items: list[WatchlistCenterItemData] = Field(default_factory=list)


class WatchlistCenterOverviewData(BaseModel):
    generated_at: datetime
    available: bool
    empty_message: str | None = None
    summary: WatchlistCenterSummaryData
    items: list[WatchlistCenterItemData] = Field(default_factory=list)
    grouped: WatchlistCenterGroupedData
